docs/tutorial/tailable-cursor.txt

```diff
@@ -13,67 +13,75 @@ Tailable Cursor
+interface, which allows it to be iterated with ``foreach`` and interface with
+any PHP functions that work with :php:`iterables <types.iterable>`. Similar to
+result objects in other database drivers, cursors in MongoDB only support
+forward iteration, which means they cannot be rewound or used with ``foreach``
+multiple times.
+
+:manual:`Tailable cursors </core/tailable-cursors>` are a special type of
+MongoDB cursor that allows the client to read some results and then wait until
+more documents become available. These cursors are primarily used with
+:manual:`Capped Collections </core/capped-collections>` and
+:manual:`Change Streams </changeStreams>`.
+
+While normal cursors can be iterated once with ``foreach``, that approach will
+not work with tailable cursors. When ``foreach`` is used with a tailable cursor,
+the loop will stop upon reaching the end of the initial result set. Attempting
+to continue iteration on the cursor with a second ``foreach`` would throw an
+exception, since PHP attempts to rewind the cursor.
+
+In order to continuously read from a tailable cursor, we will need to wrap the
+Cursor object with an :php:`IteratorIterator <iteratoriterator>`. This will
+allow us to directly control the cursor's iteration (e.g. call ``next()``),
+avoid inadvertently rewinding the cursor, and decide when to wait for new
+results or stop iteration entirely.
+
+Wrapping a Normal Cursor
+------------------------
+
+Before looking at how a tailable cursor can be wrapped with
+:php:`IteratorIterator <iteratoriterator>`, we'll start by examining how the
+class interacts with a normal cursor.
+
+The following example finds five restaurants and uses ``foreach`` to view the
+results:
 
 .. code-block:: php
 
    <?php
 
    $collection = (new MongoDB\Client)->test->restaurants;
 
-   $cursor = $collection->find(
-      [
-         'cuisine' => 'Italian',
-         'borough' => 'Manhattan',
-      ],
-      [
-         'limit' => 5,
-         'projection' => [
-            'name' => 1,
-            'borough' => 1,
-            'cuisine' => 1,
-         ],
-      ]
-   );
-
-Now we can view the results with a :code:`foreach`.
+   $cursor = $collection->find([], ['limit' => 5]);
 
-.. code-block:: php
-
-   foreach (new IteratorIterator($cursor) as $document) {
+   foreach ($cursor as $document) {
       var_dump($document);
    }
 
-The IteratorIterator equivalent of the above :code:`foreach` follows.
+While this example is quite concise, there is actually quite a bit going on. The
+``foreach`` construct begins by rewinding the iterable (``$cursor`` in this
+case). It then checks if the current position is valid. If the position is not
+valid, the loop ends. Otherwise, the current key and value are accessed
+accordingly and the loop body is executed. Assuming a :php:`break <break>` has
+not occurred, the iterator then advances to the next position, control jumps
+back to the validity check, and the loop continues.
```

Comment:
I recall we spoke about this a few months ago when I was introducing iterators, but I likely did a poor job explaining it in person. Hopefully this is easy to digest.

Reply:
This makes a lot of sense!